src/shogun/mathematics/linalg/LinalgBackendEigen.h

```diff
@@ -381,6 +391,23 @@ class LinalgBackendEigen : public LinalgBackendBase
 		return x;
 	}
 
+	/** Eigen3 QR solver */
+	template <typename T>
+	SGMatrix<T> qr_solver_impl(const SGMatrix<T>& A, const SGMatrix<T>& b) const
+	{
+		SGMatrix<T> x = SGMatrix<T>(b.num_rows, b.num_cols);
+		x.zero();
+
+		typename SGMatrix<T>::EigenMatrixXtMap A_eig = A;
+		typename SGMatrix<T>::EigenMatrixXtMap b_eig = b;
+		typename SGMatrix<T>::EigenMatrixXtMap x_eig = x;
+
+		Eigen::HouseholderQR<Eigen::Matrix<T,Eigen::Dynamic,Eigen::Dynamic> > qr(A_eig);
```

Comment:
whoooooooooo although enum seems like a super simple solution, but i reckon having a separate function for it would be better.... think about how the function that handles the enum will grown and grow and grow :)